saleor/plugins/webhook/utils.py

```diff
@@ -289,3 +297,146 @@ def get_meta_code_key(app: App) -> str:
 
 def get_meta_description_key(app: App) -> str:
     return f"{app.identifier}.description"
+
+
+def generate_cache_key_for_webhook(
+    key_data: dict, webhook_url: str, event: str, app_id: int
+) -> str:
+    """Generate cache key for webhook.
+
+    Cache key takes into account the webhook url, event type, and app id.
+    The response from webhook_url can be different for different events.
+    Apps can have assigned different permissions, so the response can vary for
+    different apps.
+    """
+    key = json.dumps(key_data)
+    return (
+        f"{app_id}-{webhook_url}-{event}-"
+        f"{hashlib.sha256(key.encode('utf-8')).hexdigest()}"
+    )
+
+
+def get_credit_card_info(
+    app: App, credit_card_info: dict
+) -> Optional[PaymentMethodCreditCardInfo]:
+    required_fields = [
+        "brand",
+        "lastDigits",
+        "expYear",
+        "expMonth",
+    ]
+    brand = credit_card_info.get("brand")
+    last_digits = credit_card_info.get("lastDigits")
+    exp_year = credit_card_info.get("expYear")
+    exp_month = credit_card_info.get("expMonth")
+    first_digits = credit_card_info.get("firstDigits")
+    if not all(field in credit_card_info for field in required_fields):
+        logger.warning(
+            "Skipping stored payment method. Missing required fields for credit card "
+            "info. Required fields: %s, received fields: %s from app %s.",
+            required_fields,
+            credit_card_info.keys(),
+            app.id,
+        )
+        return None
+    if not all([brand, last_digits, exp_year, exp_month]):
+        logger.warning("Skipping stored credit card info without required fields")
+        return None
+    if not isinstance(exp_year, int):
+        if isinstance(exp_year, str) and exp_year.isdigit():
+            exp_year = int(exp_year)
+        else:
+            logger.warning(
+                "Skipping stored payment method with invalid expYear, "
+                "received from app %s",
+                app.id,
+            )
+            return None
+
+    if not isinstance(exp_month, int):
+        if isinstance(exp_month, str) and exp_month.isdigit():
+            exp_month = int(exp_month)
+        else:
+            logger.warning(
+                "Skipping stored payment method with invalid expMonth, "
+                "received from app %s",
+                app.id,
+            )
+            return None
+
+    return PaymentMethodCreditCardInfo(
+        brand=str(brand),
+        last_digits=str(last_digits),
+        exp_year=exp_year,
+        exp_month=exp_month,
+        first_digits=str(first_digits) if first_digits else None,
+    )
+
+
+def get_payment_method_from_response(
```

Comment:
Maybe it's time to extract payment-related utils to a separate file and not clutter `utils.py` anymore, there is everything now from general utils to functions specific to particular webhooks.